### PR TITLE
Add visibilityChanged signal to debugger

### PR DESCRIPTION
**Pull request by Dovyski**
Add the `visibilityChanged` signal to the `DebuggerFrontEnd` following the same convention/idea of the already existing `drawDebugChanged` signal.

### CODE REVIEW

File: flixel/system/frontEnds/DebuggerFrontEnd.hx [hunk 1]
```diff
@@ -35,6 +35,10 @@ class DebuggerFrontEnd
 	 * Dispatched when drawDebug is changed.
 	 */
 	public var drawDebugChanged(default, null):FlxSignal = new FlxSignal();
+	/**
+	 * Dispatched when visibility is changed.
+	 */
+	public var visibilityChanged(default, null):FlxTypedSignal<Bool->Void> = new FlxTypedSignal<Bool->Void>();
```
Comment (Gama11) [inline flixel/system/frontEnds/DebuggerFrontEnd.hx]: Is it necessary to include the new value here (considering we don't do that for `drawDebugChanged` either)?

Comment (Dovyski) [inline flixel/system/frontEnds/DebuggerFrontEnd.hx]: No, it's not necessary, I was just trying to make things easier for the signal's user. I can remove the new value for the sake of consistency.

File: flixel/system/frontEnds/DebuggerFrontEnd.hx [hunk 2]
```diff
@@ -155,8 +159,11 @@ class DebuggerFrontEnd
 			FlxG.stage.stageFocusRect = false; // don't show yellow focus rect on flash
 			FlxG.stage.focus = FlxG.game;
 		}
+		
+		visible = Value;
+		visibilityChanged.dispatch();
 		#end
 		
-		return visible = Value;
```
Comment (Gama11) [inline flixel/system/frontEnds/DebuggerFrontEnd.hx]: This now leads to `set_visible()` not doing anything with `FLX_NO_DEBUG`. I think we should be consistent with the behavior of `drawDebug` here again.

File: flixel/system/frontEnds/DebuggerFrontEnd.hx [hunk 3]
```diff
@@ -155,8 +159,11 @@ class DebuggerFrontEnd
 			FlxG.stage.stageFocusRect = false; // don't show yellow focus rect on flash
 			FlxG.stage.focus = FlxG.game;
 		}
+		
+		visible = Value;
+		visibilityChanged.dispatch();
```
Comment (Gama11) [inline flixel/system/frontEnds/DebuggerFrontEnd.hx]: We shouldn't always dispatch the signal, only when the value actually _changed_.

File: flixel/system/frontEnds/DebuggerFrontEnd.hx (hunk 2 above)
Comment (Dovyski) [inline flixel/system/frontEnds/DebuggerFrontEnd.hx]: Yep, I see it now. I will fix it as soon as I have some free time.

File: flixel/system/frontEnds/DebuggerFrontEnd.hx (hunk 3 above)
Comment (Dovyski) [inline flixel/system/frontEnds/DebuggerFrontEnd.hx]: Ok, I will change that.

**Pull request merged**